src/librustc/infer/lexical_region_resolve/graphviz.rs

```diff
@@ -124,7 +124,7 @@ pub fn maybe_print_constraints_for<'a, 'gcx, 'tcx>(
 struct ConstraintGraph<'a, 'gcx: 'a+'tcx, 'tcx: 'a> {
     graph_name: String,
     region_rels: &'a RegionRelations<'a, 'gcx, 'tcx>,
-    map: &'a BTreeMap<Constraint<'tcx>, SubregionOrigin<'tcx>>,
+    map: &'a LazyBTreeMap<Constraint<'tcx>, SubregionOrigin<'tcx>>,
```

Comment:
Could this even be an `FxHashMap`?

Reply:
If `ConstraintGraph::map` is changed to an FxHashMap, then all the other LazyBTreeMap instances in this patch have to change to an FxHashMap as well. Maybe that's what you're suggesting? The various BTreeMaps are iterated over in multiple places. I don't know if the iteration order is important, but I'm inclined to be conservative.

Reply:
Yes, `Constraint` just seems to have a derived `Ord` implementation, so the order might not be important. I don't know this code though.